blueocean-dashboard/src/main/js/api/Branch.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * Simple pipeline branch API component.
+ * <p>
+ * Non-react component that contains general API methods for
+ * interacting with pipeline branches, encapsulating REST API calls etc.
+ */
+
+import fetch from 'isomorphic-fetch';
+import config from '../config';
+import Pipeline from './Pipeline';
+import * as sse from '@jenkins-cd/sse-gateway';
+import * as pushEventUtil from '../util/push-event-util';
+
+export default class Branch {
+
+    constructor(pipeline, name) {
+        this.pipeline = pipeline;
+        this.name = name;
+        this.sseListeners = [];
+    }
+
+    runDetailsRouteUrl(runId) {
+        if (runId === undefined) {
+            throw new Error('Branch.runDetailsRouteUrl must be supplied with a "runId" parameter.');
+        }
+        return `/organizations/${this.pipeline.organization}/${this.pipeline.name}/detail/${this.name}/${runId}/pipeline`;
+    }
```

Comment:
I added UrlUtils.buildRunDetailsUrl in a past PR to handle this so it would be good to use it here too. I'm working on adding more methods to it so we can centralize URL building logic. Lack of URI encoding has caused a few bugs before.